### ISSUE TITLE
Gauntlet Dark Legacy. In-game in latest builds (April 2017).

### ISSUE THREAD

**Issue by PatrickvL** [Cxbx-Reloaded/game-compatibility#142]
_From @ObiKKa on April 9, 2017 19:1_

Thanks to PR #331 (Apr 8-9, 2017?) & #334 (Apr 9-10?) I can see several stages of intro screens, a long and lovely cinematic movie, the menus and finally in-game graphics in its currently broken status. Others are welcome to test this game in the future.

### Interesting notes:
- The intro movie (CG-animated) is rendered on only a quarter of the screen at the moment. The same is true with another movie when you exit in-game back to the main menu.
- The tutorial cutout screens in-game are garbled so I cannot read anything, therefore do not know what to do in the first level.
- Many effects and shading stuff are missing (Shown as garbled). The menus are not fully coloured and look a bit corrupted.

![2017-04-10_ 01_boot](https://cloud.githubusercontent.com/assets/18428108/24840035/204eca3a-1da9-11e7-86e4-c168f0e8931a.png)
![2017-04-10_ 02_intro-movie](https://cloud.githubusercontent.com/assets/18428108/24840036/20542656-1da9-11e7-9229-ec2d0e5554b9.png)
![2017-04-10_ 03_title-screen_in-movie](https://cloud.githubusercontent.com/assets/18428108/24840046/42fb99e6-1da9-11e7-8199-ec2224ef123b.png)
![2017-04-10_ 04_title-screen](https://cloud.githubusercontent.com/assets/18428108/24840047/42fdb910-1da9-11e7-8087-2020e13cb7df.png)
![2017-04-10_ 05_start-game-profile](https://cloud.githubusercontent.com/assets/18428108/24840048/4302c9c8-1da9-11e7-816c-f0ed30d5d14f.png)
![2017-04-10_ 06_in-game_tute-screens_broken](https://cloud.githubusercontent.com/assets/18428108/24840049/4304e910-1da9-11e7-8ef6-33261458b453.png)
![2017-04-10_ 07_in-game](https://cloud.githubusercontent.com/assets/18428108/24840050/43058546-1da9-11e7-8192-8b02c1c82d76.png)

I tried to get in-game as quickly as possible on the second go to keep the kernel file small but then had to trim off much of the fluff.
[Xbe_(2017-04-10).txt](https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/files/908464/Xbe_.2017-04-10.txt)
[KrnlDebug_(2017-04-10_Trimmed).txt](https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/files/908465/KrnlDebug_.2017-04-10_Trimmed.txt)

_Copied from original issue: Cxbx-Reloaded/Cxbx-Reloaded#335_

**Comment by PatrickvL** [comment]
_From @ObiKKa on April 9, 2017 19:5_

Ah, that was then (6.5 years ago). Wow they look virtually similar! 
http://shogun3d-cxbx.blogspot.com.au/2010/11/gauntlet-dark-legacy.html

**Comment by PatrickvL** [comment]
_From @BenNottelling on April 9, 2017 19:10_

It's good they are similar, that means the emulator hadn't really made any backwards steps

**Comment by PatrickvL** [comment]
Dxbx, back in 2010 : https://www.youtube.com/watch?v=NWlmKP74LuY

**Comment by PatrickvL** [comment]
To get menu's and in game graphics in a readable state would require better support for palletized textures then what we have right now. The Dxbx run I linked to above, used an approach that we can strife for in Cxbx too: conversion of textures at drawing time (as that's the most suitable moment to combine active palettes with active textures)

**Comment by PatrickvL** [comment]
_From @ObiKKa on April 9, 2017 21:16_

Yes. I've just now taken a look at that channel's old recorded DXBX videos from nearly 6.5 years ago.
Here's also the intro, menu and game video of that game:
https://www.youtube.com/watch?v=27mnsM-n-GU
It was uploaded 2 days before the above video which fixed stuff.

**Comment by PatrickvL** [comment]
Yeah, it's my channel so I know about that ;)

**Comment by haekb** [comment]
Small regression, not sure when it was introduced. Here's hopefully relevant info!
[Xbe.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/3389506/Xbe.txt)
[KrnlDebug.zip](https://github.com/Cxbx-Reloaded/game-compatibility/files/3389507/KrnlDebug.zip)
![gauntlet-failed-to-create-vertex-buffer](https://user-images.githubusercontent.com/3683240/61179881-f7ba5b00-a5c0-11e9-91b5-1acdd19610ba.jpg)

Also running this on debug mode in VS2017 with remote debugging tool installed generates multiple access violations pretty much every second.

**Comment by RadWolfie** [comment]
> Also running this on debug mode in VS2017 with remote debugging tool installed generates multiple access violations pretty much every second.

@haekb Multiple access violations every seconds are actually handled by LLE by the way. We're using Windows' exception handler to handle the LLE code. You can ignore it for now.

**Comment by Kalen6** [comment]
Hey there, I'm entirely new to Xbox emulation and CXBX, but they've been very interesting to familiarize myself with. It's almost July 2020, and upon attempting to emulate my Gauntlet: Dark Legacy .xbe file, it would seemingly work perfectly for about 10-20 seconds, before displaying "Received Fatal Message: Failed to create vertex buffer", and then force-closing the emulation. I'm on Windows 10, Intel processor, AMD graphics. Please let me know if there is a solution to this crash, or any way I can assist the community on this issue. Cheers!
![gauntlet error](https://user-images.githubusercontent.com/67610462/86060683-7d2d5a00-ba22-11ea-9568-7ad2b6382e18.png)

**Comment by PatrickvL** [comment]
The same issue was reported a few months back, but thanks for re-testing this. (I asked @LukeUsher to look into the vertex buffer cache, which seems to be overflowing.)

**Comment by Lethalbanana626** [comment]
So I see a lot of people have gotten this game on cxbx, how and where can I go to get this game?? I use to love this game on the xbox but I can't find any website that has an xbe file download? Where do I go???

**Comment by Lethalbanana626** [comment]
Yeah when will that fatal error be fixed?? I just want to know so I can start playing it as soon as possible cause I just downloaded it!

**Comment by Margen67** [comment]
@Lethalbanana626 Piracy isn't allowed.

**Comment by Lethalbanana626** [comment]
Im having the same issue, plz let me know whenever its fixed!

**Comment by Lethalbanana626** [comment]
Hi i noticed an issue, I dont know if its just me but whenever i load the game, the intro plays just fine, but once I get to the start screen my frames drop tremendously and it makes it basically unplayable. I just wanted to bring light to this issue because on the compatability list is says it is playable but for me it is not. I also tried another game to see if I had issues with framerate but it just seems to be this game. Help plz

**Comment by PatrickvL** [comment]
The slowness during gameplay is a known issue : it's because this game uses palletised textures, which aren't available on Windows, so we have to convert them. This takes a lot of processing power, not because of the conversation itself (which is rather trivial), but because of the continuous hashing to test whether a texture needs to be updated.

I've tried performing the palette lookup in the pixel shader, which kinda works but produces artifacts that can't be solved easily. So we're stuck with the current approach for now

**Comment by Lethalbanana626** [comment]
Ok, well I hope we'll be able to find a solution for it in the near future. thanks for the response!
